src/otaclient/ota_core.py

```diff
@@ -379,161 +494,69 @@ def _download_resources(self, resource_meta: ResourceMeta) -> None:
             self._downloader_pool.release_all_instances()
             self._downloader_pool.shutdown()
 
-    def _process_persistents(self, ota_metadata: OTAMetadata):
-        logger.info("start persist files handling...")
-        standby_slot_mp = Path(cfg.STANDBY_SLOT_MNT)
-
-        _handler = PersistFilesHandler(
-            src_passwd_file=Path(cfg.PASSWD_FPATH),
-            src_group_file=Path(cfg.GROUP_FPATH),
-            dst_passwd_file=Path(standby_slot_mp / "etc/passwd"),
-            dst_group_file=Path(standby_slot_mp / "etc/group"),
-            src_root=cfg.ACTIVE_SLOT_MNT,
-            dst_root=cfg.STANDBY_SLOT_MNT,
+    def _downloader_workder_initializer(self) -> None:
+        self._downloader_mapper[threading.get_native_id()] = (
+            self._downloader_pool.get_instance()
         )
 
-        for persiste_entry in ota_metadata.iter_persist_entries():
-            # NOTE(20240520): with update_swapfile ansible role being used wildly,
-            #   now we just ignore the swapfile entries in the persistents.txt if any,
-            #   and issue a warning about it.
-            if persiste_entry in ["/swapfile", "/swap.img"]:
-                logger.warning(
-                    f"swapfile entry {persiste_entry} is listed in persistents.txt, ignored"
-                )
-                logger.warning(
-                    (
-                        "using persis file feature to preserve swapfile is MISUSE of persist file handling feature!"
-                        "please change your OTA image build setting and remove swapfile entries from persistents.txt!"
-                    )
-                )
-                continue
 
-            try:
-                _handler.preserve_persist_entry(persiste_entry)
-            except Exception as e:
-                _err_msg = f"failed to preserve {persiste_entry}: {e!r}, skip"
-                logger.warning(_err_msg)
+class _OTAUpdater(_OTAUpdateOperator):
+    """The implementation of OTA update logic."""
 
-    def _download_and_parse_metadata(self) -> None:
-        self._download_watchdog_ctx["previous_active_timestamp"] = int(time.time())
-        _mapper = ThreadPoolExecutorWithRetry(
-            max_concurrent=cfg.MAX_CONCURRENT_DOWNLOAD_TASKS,
-            max_workers=cfg.DOWNLOAD_THREADS,
-            thread_name_prefix="download_metadata_files",
-            initializer=self._downloader_workder_initializer,
-            watchdog_func=partial(
-                self._downloader_pool.downloading_watchdog,
-                ctx=self._download_watchdog_ctx,
-                max_idle_timeout=cfg.DOWNLOAD_INACTIVE_TIMEOUT,
-            ),
+    def __init__(
+        self,
+        *,
+        version: str,
+        raw_url_base: str,
+        cookies_json: str,
+        ca_chains_store: CAChainStore,
+        upper_otaproxy: str | None = None,
+        boot_controller: BootControllerProtocol,
+        create_standby_cls: type[RebuildMode],
```

Comment:
In this existing data update class, use the existing `__init__` arguments. In other words, it includes some data update dedicated inputs like `boot_controller` and `create_standby_cls`.